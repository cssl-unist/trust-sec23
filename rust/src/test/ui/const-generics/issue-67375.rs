// revisions: full min

#![cfg_attr(full, allow(incomplete_features))]
#![cfg_attr(full, feature(const_generics))]
#![cfg_attr(min, feature(min_const_generics))]

struct Bug<T> {
    //~^ ERROR parameter `T` is never used
    inner: [(); { [|_: &T| {}; 0].len() }],
    //[min]~^ ERROR generic parameters may not be used in const operations
    //[full]~^^ WARN cannot use constants which depend on generic parameters in types
    //[full]~^^^ WARN this was previously accepted by the compiler
}

fn main() {}
