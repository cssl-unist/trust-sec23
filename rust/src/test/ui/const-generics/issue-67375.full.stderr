warning: cannot use constants which depend on generic parameters in types
  --> $DIR/issue-67375.rs:9:12
   |
LL |     inner: [(); { [|_: &T| {}; 0].len() }],
   |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(const_evaluatable_unchecked)]` on by default
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #76200 <https://github.com/rust-lang/rust/issues/76200>

error[E0392]: parameter `T` is never used
  --> $DIR/issue-67375.rs:7:12
   |
LL | struct Bug<T> {
   |            ^ unused parameter
   |
   = help: consider removing `T`, referring to it in a field, or using a marker such as `PhantomData`

error: aborting due to previous error; 1 warning emitted

For more information about this error, try `rustc --explain E0392`.
